test/groovy/com/sap/piper/TemporaryCredentialsUtilsTest.groovy

```diff
@@ -0,0 +1,136 @@
+    def bodyExecuted
+    TemporaryCredentialsUtils credUtils
+
+    @Rule
+    public RuleChain ruleChain = Rules
+        .getCommonRules(this)
+        .around(thrown)
+        .around(readYamlRule)
+        .around(credentialsRule)
+        .around(loggingRule)
+        .around(writeFileRule)
+        .around(fileExistsRule)
+        .around(readFileRule)
+        .around(shellRule)
+
+    @Before
+    void init() {
+        bodyExecuted = false
+
+        credentialsRule.reset()
+            .withCredentials('erp-credentials', 'test_user', '********')
+            .withCredentials('testCred2', 'test_other', '**')
+
+        credUtils = new TemporaryCredentialsUtils(nullScript)
+    }
+
+    @Test
+    void credentialsFileWritten() {
+        def credential = [alias: 'ERP', credentialId: 'erp-credentials']
+        def directory = './'
+        def filename = 'credentials.json'
+        fileExistsRule.registerExistingFile('systems.yml')
+
+        credUtils.writeCredentials([credential], directory, filename )
+
+        assertThat(writeFileRule.files['credentials.json'], containsString('"alias":"ERP","username":"test_user","password":"********"'))
+    }
+
+    @Test
+    void credentialsFileNotWrittenWithEmptyList() {
+        def directory = './'
+        def filename = 'credentials.json'
+        fileExistsRule.registerExistingFile('systems.yml')
+
+        credUtils.writeCredentials([], directory, filename )
+
+        loggingRule.expect('Not writing any credentials.')
+    }
+
+    @Test
+    void credentialsFileDeleted() {
+        def directory = './'
+        def filename = 'credentials.json'
+        fileExistsRule.registerExistingFile('systems.yml')
+
+        credUtils.deleteCredentials(directory, filename )
+
+        assertThat(shellRule.shell, hasItem('rm -f credentials.json'))
+    }
+
+    @Test
+    void systemsFileNotExists() {
+        def directory = './'
+        thrown.expect(hudson.AbortException)
+        thrown.expectMessage("The directory ${directory} does not contain any of the files systems.yml, systems.yaml or systems.json. " +
+            "One of those files is required in order to activate the integration test credentials configured in the pipeline configuration file of this project. " +
+            "Please add the file as explained in the SAP Cloud SDK documentation.")
+
+        credUtils.assertSystemsFileExists(directory)
+    }
+
+    @Test
+    void handleTemporaryCredentials() {
+        def credential = [alias: 'ERP', credentialId: 'erp-credentials']
+        def directory = './'
+        fileExistsRule.registerExistingFile('systems.yml')
+
+        credUtils.handleTemporaryCredentials([credential], directory) {
+            bodyExecuted = true
+        }
+        assertTrue(bodyExecuted)
+        assertThat(writeFileRule.files['credentials.json'], containsString('"alias":"ERP","username":"test_user","password":"********"'))
+        assertThat(shellRule.shell, hasItem('rm -f credentials.json'))
+    }
```

Comment:
Hope it's OK: I would be interested in one more test where you call handleTemporaryCredentials() with an empty list and assert the reverse. Background: I am not sure if assigning a boolean variable uses Groovy Truth just like it would work in an if-condition.

Reply:
Isn't that exactly what is tested [here]( https://github.com/SAP/jenkins-library/blob/temp-credentials/test/groovy/com/sap/piper/TemporaryCredentialsUtilsTest.groovy#L128)?

Reply:
Yes, and maybe I should have more coffee.